uds/uds_config_tool/SupportedServices/ReadDataByIdentifierContainer.py

```diff
@@ -21,56 +23,25 @@ class ReadDataByIdentifierContainer(object):
 
     def __init__(self):
 
-        # To cater for lists we may have to re-factor here - i.e. requestFunc can be split into requestSIDFunc and requestDIDFunc to allow building on the fly from a DID list
-        # Also checkFunction into: checkSIDResonseFunction+SIDLengthFunction, checkResponseLengthFunction + responseLengthFunction, and an iterable checkDIDResponseFunction
-        # Also positiveResponseFunc into: positiveResponseSIDFunction, and an iterable positiveResponseDIDFunction
+        # To cater for lists we may have to re-factor here - i.e. requestFunc can be split into
+        # requestSIDFunc and requestDIDFunc to allow building on the fly from a DID list
         # Negative response function is ok as it it
 
-        # self.requestFunctions = {}
         self.requestSIDFunctions = {}
         self.requestDIDFunctions = {}
 
-        # self.checkFunctions = {}
-        self.checkSIDResponseFunctions = {}
-        self.checkSIDLengthFunctions = {}
-        self.checkDIDResponseFunctions = {}
-        self.checkDIDLengthFunctions = {}
+        self.posResponseObjects = {}
 
         self.negativeResponseFunctions = {}
 
-        self.positiveResponseFunctions = {}
-
     ##
     # @brief this method is bound to an external Uds object so that it call be called
     # as one of the in-built methods. uds.readDataByIdentifier("something") It does not operate
     # on this instance of the container class.
     @staticmethod
     def __readDataByIdentifier(target, parameter):
-        # Some local functions to deal with use concatenation of a number of DIDs in RDBI operation ...
-
-        # After an array of lengths has been constructed for the individual response elements, we need a simple function to check it against the response
-        def checkTotalResponseLength(input, expectedLengthsList):
-            lengthExpected = sum(expectedLengthsList)
-            if len(input) != lengthExpected:
-                raise Exception(
-                    "Total length returned not as expected. Expected: {0}; Got {1}".format(
-                        lengthExpected, len(input)
-                    )
-                )
-
-        # The check functions just want to know about the next bit of the response, so this just pops it of the front of the response
-        def popResponseElement(input, expectedList):
-            if expectedList == []:
-                raise Exception(
-                    "Total length returned not as expected. Missing elements."
-                )
-            return (
-                input[0 : expectedList[0]],
-                input[expectedList[0] :],
-                expectedList[1:],
-            )
-
-        dids = parameter
+
+        dids: str | List[str] = parameter
```

Comment:
Except by adding `from __future__ import annotations` at the very top of the module, this isn't python 3.7/3.8 compliant

If you add this import you can even use:
`dids: str | list[str] = parameter`

Reply:
fixed